tests/functional/aws-node-sdk/test/object/rangeTest.js

```diff
@@ -64,20 +64,28 @@ function checkRanges(range, bytes) {
 }
 
 // Create 5MB parts and upload them as parts of a MPU
-function uploadParts(bytes, uploadId) {
+async function uploadParts(bytes, uploadId) {
     const name = `hashedFile.${bytes}`;
-
-    return Promise.map([1, 2], part =>
-        execFileAsync('dd', [`if=${name}`, `of=${name}.mpuPart${part}`,
-            'bs=5242880', `skip=${part - 1}`, 'count=1'])
-        .then(() => s3.uploadPart({
-            Bucket: bucket,
-            Key: key,
-            PartNumber: part,
-            UploadId: uploadId,
-            Body: createReadStream(`${name}.mpuPart${part}`),
-        }).promise())
-    );
+    for (let part = 1; part <= 2; part++) {
+        try {
+            await execFileAsync('dd', [
+                `if=${name}`,
+                `of=${name}.mpuPart${part}`,
+                'bs=5242880', 
+                `skip=${part - 1}`,
+                'count=1',
+            ]);
+            await s3.uploadPart({
+                Bucket: bucket,
+                Key: key,
+                PartNumber: part,
+                UploadId: uploadId,
+                Body: createReadStream(`${name}.mpuPart${part}`),
+            }).promise();
+        } catch (error) {
+            throw new Error(`Error uploading part ${part}: ${error.message}`);
+        }
+    }
```

Comment:
this is actually not the same, as the 2 parts which used to be uploaded in parallel are now send sequentially.

the equivalient behavior can be achieved with `Promise.all([1, 2].map(...))` (or `Promise.allSettled()`):
```suggestion
    return Promise.all([1, 2].map(async part => {
        try {
            await execFileAsync('dd', [
                `if=${name}`,
                `of=${name}.mpuPart${part}`,
                'bs=5242880', 
                `skip=${part - 1}`,
                'count=1',
            ]);
            await s3.uploadPart({
                Bucket: bucket,
                Key: key,
                PartNumber: part,
                UploadId: uploadId,
                Body: createReadStream(`${name}.mpuPart${part}`),
            }).promise();
        } catch (error) {
            throw new Error(`Error uploading part ${part}: ${error.message}`);
        }
    }));
```

not sure if this is relevant or really affects the tests...

Reply:
I don't think this helps with test coverage, but I applied the suggestion.